src/backend/chat/keepalive.py

```diff
@@ -0,0 +1,156 @@
+"""Helpers to prevent proxy timeouts during long-running stream operations.
+
+This module provides utilities to wrap synchronous and asynchronous iterators
+with keepalive messages. When a stream pauses for longer than the specified
+interval, keepalive messages are injected to prevent proxy/gateway
+timeouts while waiting for the stream data.
+"""
+
+import asyncio
+import logging
+import queue
+import threading
+import time
+from typing import AsyncIterator, Iterator
+
+logger = logging.getLogger(__name__)
+
+# Default keepalive message sent to client during stream pauses
+KEEPALIVE_MESSAGE = '2:[{"status":"WAITING"}]\n'
```

Comment:
This message seems highly related to the vercel frontend SDK, is it required to be? I mean: what would happen if this is actually an empty string, or a simple space (` `) or a simple character? would it break the front-end ? 
Should we consider using the vercel `EventEncoder` ? I'm just afraid it breaks when upgrading vercel version later.